src/serializer/ResidualHeapVisitor.js

```diff
@@ -487,30 +495,90 @@ export class ResidualHeapVisitor {
         this.referencedDeclaredValues.add(value);
       },
       recordDelayedEntry: (entry: GeneratorEntry) => {
-        this.delayedVisitGeneratorEntries.push({ generator, entry });
+        this.delayedVisitGeneratorEntries.push({ baseContext, generator, entry });
       },
     };
   }
 
-  visitGenerator(generator: Generator): void {
+  visitGenerator(generator: Generator, afterGeneratorVisit?: () => void): void {
```

Comment:
Is there a deep reason for the new `afterGeneratorVisit` complication? Below where this is being used, I would just issue another `this._withScope` call.